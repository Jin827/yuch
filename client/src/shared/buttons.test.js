import React from 'react';
import { shallow } from 'enzyme';
import { unwrap } from '@material-ui/core/test-utils';
import Buttons from './buttons';

describe('material ui button', () => {
  const UnrappedButton = unwrap(Buttons);
  const handleFirstButtonClick = jest.fn();
  const handleSecondButtonClick = jest.fn();

  const setup = () => {
    const props = {
      classes: {
        button: {
          width: '6em',
          paddingTop: '5px',
          paddingBottom: '5px',
        },
      },

      firstButtonName: '예약하기',
      secondButtonName: '가입하기',
      handleFirstButtonClick,
      handleSecondButtonClick,
    };
    const wrapper = shallow(<UnrappedButton {...props} />);
    return { props, wrapper };
  };

  const { wrapper } = setup();

  describe('first button', () => {
    it('display button correctly', () => {
      expect(wrapper.find('.firstBtn').exists()).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });

    it('click event should exists', () => {
      const ev = {};
      wrapper.find('.firstBtn').simulate('click', ev);
      expect(handleFirstButtonClick.mock.calls).toHaveLength(1);
      expect(handleFirstButtonClick).toHaveBeenCalled();
    });
  });
  describe('second button', () => {
    it('display button correctly', () => {
      expect(wrapper.find('.secBtn').exists()).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });

    it('click event should exists', () => {
      const ev = {};
      wrapper.find('.secBtn').simulate('click', ev);
      expect(handleSecondButtonClick.mock.calls).toHaveLength(1);
      expect(handleSecondButtonClick).toHaveBeenCalled();
    });
  });
});
