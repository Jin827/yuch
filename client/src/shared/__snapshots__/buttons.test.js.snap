// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`material ui button first button display button correctly 1`] = `
<Fragment>
  <WithStyles(Button)
    className="firstBtn [object Object]"
    color="secondary"
    onClick={[Function]}
    variant="contained"
  >
    예약하기
  </WithStyles(Button)>
  <WithStyles(Button)
    className="secBtn [object Object]"
    color="secondary"
    onClick={[Function]}
    variant="outlined"
  >
    가입하기
  </WithStyles(Button)>
</Fragment>
`;

exports[`material ui button second button display button correctly 1`] = `
<Fragment>
  <WithStyles(Button)
    className="firstBtn [object Object]"
    color="secondary"
    onClick={[Function]}
    variant="contained"
  >
    예약하기
  </WithStyles(Button)>
  <WithStyles(Button)
    className="secBtn [object Object]"
    color="secondary"
    onClick={[Function]}
    variant="outlined"
  >
    가입하기
  </WithStyles(Button)>
</Fragment>
`;
